source/common/common/utility.h

```diff
@@ -288,4 +301,59 @@ class RegexUtil {
                                std::regex::flag_type flags = std::regex::optimize);
 };
 
+/**
+ * Maintains sets of numeric intervals. As new intervals are added, existing ones in the
+ * set are combined so that no overlapping intervals remain in the representation.
+ *
+ * Value can be any type that is comparable with <.
+ */
+template <typename Value> class IntervalSetImpl : public IntervalSet<Value> {
+public:
+  typedef typename IntervalSet<Value>::Interval Interval;
+
+  // Inserts a new interval into the set, merging any overlaps.
+  void insert(Value left, Value right) override {
+    if (left == right) {
```

Comment:
Can you document that the intervals are exclusive (as opposed to inclusive) in the base class so it's clear to users?

Reply:
done